From df653a720a90b25f6e890bc0e821c9c75cca49e4 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 11 Jan 2016 17:42:44 +0800
Subject: [PATCH 0828/5242] MLK-12188 dmaengine: fsl-edma: add fsl_chan->edesc
 NULL check

commit  b094f8daff5251d32ceaba495d3b09aa1ebf1b14 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add fsl_chan->edesc NULL when calling .fsl_edma_desc_residue() to avoid
NULL pointer access.

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/fsl-edma.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/fsl-edma.c b/drivers/dma/fsl-edma.c
index c756886..34a7bf6 100644
--- a/drivers/dma/fsl-edma.c
+++ b/drivers/dma/fsl-edma.c
@@ -426,7 +426,7 @@ static enum dma_status fsl_edma_tx_status(struct dma_chan *chan,
 	vdesc = vchan_find_desc(&fsl_chan->vchan, cookie);
 	if (fsl_chan->edesc && cookie == fsl_chan->edesc->vdesc.tx.cookie)
 		txstate->residue = fsl_edma_desc_residue(fsl_chan, vdesc, true);
-	else if (vdesc)
+	else if (fsl_chan->edesc && vdesc)
 		txstate->residue = fsl_edma_desc_residue(fsl_chan, vdesc, false);
 	else
 		txstate->residue = 0;
-- 
1.7.9.5

